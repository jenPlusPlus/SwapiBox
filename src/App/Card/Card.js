import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ cardData, cardType, updateFavorites, isFavorite }) => {


  if (cardData.cardType === 'people') {
    let mappedSpecies;
    if (cardData.species.length > 0) {
      mappedSpecies = cardData.species.map((species, index) => {
        return <li className='value' key={index + Date.now()}>{species}</li>;
      });
    } else {
      mappedSpecies = <span className='value'>Unknown</span>;
    }
    return (
      <div className='card'>
        <div className='card-title'>
          <h3 className='people-name card-name'>
            {cardData.name}</h3>
          <div className={`favorite ${isFavorite}`}
            onClick={updateFavorites.bind(this, cardData)}></div>
        </div>
        <div className='card-info'>
          <p className='people-homeworld label'>Homeworld:
            <span className='value'> {cardData.homeworld}</span></p>
          <div className='people-species'>
            <ul><span className='label'>Species: </span>{mappedSpecies}</ul>
          </div>
          <p className='people-homeworld-population label'>Homeworld Population:
            <span className='value'> {cardData.homeworldPopulation}</span></p>
        </div>
      </div>
    );
  } else if (cardData.cardType === 'vehicles') {
    return (
      <div className='card'>
        <div className='card-title'>
          <h3 className='vehicles-name card-name'>
            {cardData.name}</h3>
          <div className={`favorite ${isFavorite}`}
            onClick={updateFavorites.bind(this, cardData)}></div>
        </div>
        <div className='card-info'>
          <p className='vehicles-model label'>Model:
            <span className='value'> {cardData.model}</span>
          </p>
          <p className='vehicles-class label'>Class:
            <span className='value'> {cardData.class}</span>
          </p>
          <p className='vehicles-num-passengers label'>Number of Passengers:
            <span className='value'> {cardData.numPassengers}</span></p>
        </div>
      </div>
    );
  } else if (cardData.cardType === 'planets') {
    let mappedResidents;
    if (cardData.residents.length > 0) {
      mappedResidents = cardData.residents.map((resident, index) => {
        return <li className='value' key={index + Date.now()}>{resident}</li>;
      });
    } else {
      mappedResidents = <span className='value'>Unknown</span>;
    }
    return (
      <div className='card'>
        <div className='card-title'>
          <h3 className='planets-name card-name'>
            {cardData.name}</h3>
          <div className={`favorite ${isFavorite}`}
            onClick={updateFavorites.bind(this, cardData)}></div>
        </div>
        <div className='card-info'>
          <p className='planets-terrain label'>Terrain:
            <span className='value'> {cardData.terrain}</span></p>
          <p className='planets-population label'>Population:
            <span className='value'> {cardData.population}</span></p>
          <p className='planets-climate label'>Climate:
            <span className='value'> { cardData.climate}</span></p>
          <div className='planets-residents'>
            <ul><span className='label'>Residents: </span>
              <div className='residents-list-container'>{mappedResidents}
              </div>
            </ul>
          </div>
        </div>
      </div>
    );
  }
};

Card.propTypes = {
  cardData: PropTypes.object,
  cardType: PropTypes.string,
  updateFavorites: PropTypes.func,
  isFavorite: PropTypes.string
};
export default Card;
