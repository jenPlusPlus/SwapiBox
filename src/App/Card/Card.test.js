import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import fetchMock from 'fetch-mock';
import renderer from 'react-test-renderer';
import { withRouter } from 'react-router';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router, Route } from 'react-router-dom';

describe('Card', () => {
  const mkFun = jest.fn();
  let wrapper = shallow(<Card cardType={'People'}
                              cardData={{
                                cardType: "people",
                                homeworld:"Tatooine",
                                homeworldPopulation: "200000",
                                name: "Luke Skywalker",
                                species:["Human"]
                              }}
                              key={1}
                              updatedFavorites={mkFun}
                              isFavorite={''}
                              />);

it('should match the Card snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});
