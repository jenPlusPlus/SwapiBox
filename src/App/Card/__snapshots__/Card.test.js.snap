// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the Card snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardData={
        Object {
            "cardType": "people",
            "homeworld": "Tatooine",
            "homeworldPopulation": "200000",
            "name": "Luke Skywalker",
            "species": Array [
              "Human",
            ],
          }
    }
    cardType="People"
    isFavorite=""
    updatedFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-title"
>
          <h3
                    className="people-name card-name"
          >
                    Luke Skywalker
          </h3>
          <div
                    className="favorite "
                    onClick={[Function]}
          />
</div>,
        <div
          className="card-info"
>
          <p
                    className="people-homeworld label"
          >
                    Homeworld:
                    <span
                              className="value"
                    >
                               
                              Tatooine
                    </span>
          </p>
          <div
                    className="people-species"
          >
                    <ul>
                              <span
                                        className="label"
                              >
                                        Species: 
                              </span>
                              <div
                                        className="species-list-container"
                              >
                                        <li
                                                  className="value"
                                        >
                                                  Human
                                        </li>
                              </div>
                    </ul>
          </div>
          <p
                    className="people-homeworld-population label"
          >
                    Homeworld Population:
                    <span
                              className="value"
                    >
                               
                              200000
                    </span>
          </p>
</div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="people-name card-name"
>
              Luke Skywalker
</h3>,
            <div
              className="favorite "
              onClick={[Function]}
/>,
          ],
          "className": "card-title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Luke Skywalker",
              "className": "people-name card-name",
            },
            "ref": null,
            "rendered": "Luke Skywalker",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "className": "favorite ",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="people-homeworld label"
>
              Homeworld:
              <span
                            className="value"
              >
                             
                            Tatooine
              </span>
</p>,
            <div
              className="people-species"
>
              <ul>
                            <span
                                          className="label"
                            >
                                          Species: 
                            </span>
                            <div
                                          className="species-list-container"
                            >
                                          <li
                                                        className="value"
                                          >
                                                        Human
                                          </li>
                            </div>
              </ul>
</div>,
            <p
              className="people-homeworld-population label"
>
              Homeworld Population:
              <span
                            className="value"
              >
                             
                            200000
              </span>
</p>,
          ],
          "className": "card-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld:",
                <span
                  className="value"
>
                   
                  Tatooine
</span>,
              ],
              "className": "people-homeworld label",
            },
            "ref": null,
            "rendered": Array [
              "Homeworld:",
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "Tatooine",
                  ],
                  "className": "value",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "Tatooine",
                ],
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <ul>
                <span
                                className="label"
                >
                                Species: 
                </span>
                <div
                                className="species-list-container"
                >
                                <li
                                                className="value"
                                >
                                                Human
                                </li>
                </div>
</ul>,
              "className": "people-species",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="label"
>
                    Species: 
</span>,
                  <div
                    className="species-list-container"
>
                    <li
                                        className="value"
                    >
                                        Human
                    </li>
</div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Species: ",
                    "className": "label",
                  },
                  "ref": null,
                  "rendered": "Species: ",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <li
                        className="value"
>
                        Human
</li>,
                    ],
                    "className": "species-list-container",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "Luke Skywalker0",
                      "nodeType": "host",
                      "props": Object {
                        "children": "Human",
                        "className": "value",
                      },
                      "ref": null,
                      "rendered": "Human",
                      "type": "li",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld Population:",
                <span
                  className="value"
>
                   
                  200000
</span>,
              ],
              "className": "people-homeworld-population label",
            },
            "ref": null,
            "rendered": Array [
              "Homeworld Population:",
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "200000",
                  ],
                  "className": "value",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "200000",
                ],
                "type": "span",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-title"
>
            <h3
                        className="people-name card-name"
            >
                        Luke Skywalker
            </h3>
            <div
                        className="favorite "
                        onClick={[Function]}
            />
</div>,
          <div
            className="card-info"
>
            <p
                        className="people-homeworld label"
            >
                        Homeworld:
                        <span
                                    className="value"
                        >
                                     
                                    Tatooine
                        </span>
            </p>
            <div
                        className="people-species"
            >
                        <ul>
                                    <span
                                                className="label"
                                    >
                                                Species: 
                                    </span>
                                    <div
                                                className="species-list-container"
                                    >
                                                <li
                                                            className="value"
                                                >
                                                            Human
                                                </li>
                                    </div>
                        </ul>
            </div>
            <p
                        className="people-homeworld-population label"
            >
                        Homeworld Population:
                        <span
                                    className="value"
                        >
                                     
                                    200000
                        </span>
            </p>
</div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="people-name card-name"
>
                Luke Skywalker
</h3>,
              <div
                className="favorite "
                onClick={[Function]}
/>,
            ],
            "className": "card-title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Luke Skywalker",
                "className": "people-name card-name",
              },
              "ref": null,
              "rendered": "Luke Skywalker",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "className": "favorite ",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="people-homeworld label"
>
                Homeworld:
                <span
                                className="value"
                >
                                 
                                Tatooine
                </span>
</p>,
              <div
                className="people-species"
>
                <ul>
                                <span
                                                className="label"
                                >
                                                Species: 
                                </span>
                                <div
                                                className="species-list-container"
                                >
                                                <li
                                                                className="value"
                                                >
                                                                Human
                                                </li>
                                </div>
                </ul>
</div>,
              <p
                className="people-homeworld-population label"
>
                Homeworld Population:
                <span
                                className="value"
                >
                                 
                                200000
                </span>
</p>,
            ],
            "className": "card-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld:",
                  <span
                    className="value"
>
                     
                    Tatooine
</span>,
                ],
                "className": "people-homeworld label",
              },
              "ref": null,
              "rendered": Array [
                "Homeworld:",
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "Tatooine",
                    ],
                    "className": "value",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "Tatooine",
                  ],
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <ul>
                  <span
                                    className="label"
                  >
                                    Species: 
                  </span>
                  <div
                                    className="species-list-container"
                  >
                                    <li
                                                      className="value"
                                    >
                                                      Human
                                    </li>
                  </div>
</ul>,
                "className": "people-species",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="label"
>
                      Species: 
</span>,
                    <div
                      className="species-list-container"
>
                      <li
                                            className="value"
                      >
                                            Human
                      </li>
</div>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Species: ",
                      "className": "label",
                    },
                    "ref": null,
                    "rendered": "Species: ",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <li
                          className="value"
>
                          Human
</li>,
                      ],
                      "className": "species-list-container",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": "Luke Skywalker0",
                        "nodeType": "host",
                        "props": Object {
                          "children": "Human",
                          "className": "value",
                        },
                        "ref": null,
                        "rendered": "Human",
                        "type": "li",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "ul",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Homeworld Population:",
                  <span
                    className="value"
>
                     
                    200000
</span>,
                ],
                "className": "people-homeworld-population label",
              },
              "ref": null,
              "rendered": Array [
                "Homeworld Population:",
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "200000",
                    ],
                    "className": "value",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "200000",
                  ],
                  "type": "span",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
